homeassistant/components/airzone/coordinator.py

```diff
@@ -32,7 +33,7 @@ def __init__(self, hass: HomeAssistant, airzone: AirzoneLocalApi) -> None:
             update_interval=SCAN_INTERVAL,
         )
 
-    async def _async_update_data(self):
+    async def _async_update_data(self) -> Any:
```

Comment:
If I change this to `async def _async_update_data(self) -> dict[str, Any]:`, mypy complains with the following error: `homeassistant/components/airzone/coordinator.py:43: error: Returning Any from function declared to return "Dict[str, Any]"  [no-any-return]`
However, aioairzone.data() returns `dict[str, Any]`...

Reply:
```diff
diff --git a/homeassistant/components/airzone/coordinator.py b/homeassistant/components/airzone/coordinator.py
index 1856a447d3..178e320794 100644
--- a/homeassistant/components/airzone/coordinator.py
+++ b/homeassistant/components/airzone/coordinator.py
@@ -3,7 +3,7 @@ from __future__ import annotations
 
 from datetime import timedelta
 import logging
-from typing import Any
+from typing import Any, cast
 
 from aioairzone.exceptions import AirzoneError
 from aioairzone.localapi import AirzoneLocalApi
@@ -30,14 +30,15 @@ class AirzoneUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
             hass,
             _LOGGER,
             name=DOMAIN,
+            update_method=self._async_update,
             update_interval=SCAN_INTERVAL,
         )
 
-    async def _async_update_data(self) -> Any:
+    async def _async_update(self) -> dict[str, Any]:
         """Update data via library."""
         async with async_timeout.timeout(AIOAIRZONE_DEVICE_TIMEOUT_SEC):
             try:
                 await self.airzone.update()
             except AirzoneError as error:
                 raise UpdateFailed(error) from error
-            return self.airzone.data()
+            return cast(dict[str, Any], self.airzone.data())
```

Reply:
If you library is https://peps.python.org/pep-0561/ compliant you won't need the cast